### DISCUSSION TITLE
Units

### DISCUSSION THREAD

**Discussion by rl3fz** [idaholab/moose#18484] [Q&A General] [answered]
Hello everyone!
Is there a way to check the units of values in my model? Does MOOSE operate in SI units by default?
For instance, are auxkernels of type TotalFreeEnergy operating in joules?

**Comment by GiudGiud** [comment]
Should be SI units by default everywhere.
Your mesh should be in SI units too for that to be true. 

Guillaume

**Comment by dschwen** [comment, marked as answer]
The units in MOOSE depend on the parameterization of your model. Most kernels and materials expect parameters to be supplied by the user using a consistent set of units. Those do not necessarily have to be SI units at all. Some materials and kernels may have default values for certain parameters (such as the Boltzmann constant) that you can override to choose whatever unit system you deem appropriate. Here is an example:

https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/phase_field/src/materials/GasFreeEnergyBase.C#L24-L32

In this case it is easiest to just go with the default units of these quantities.

Not that we have a units system (https://mooseframework.inl.gov/source/utils/Units.html) that can be used to convert quantities inside your input file.

**Reply by dschwen** [reply]
note that _eV_ is not an SI unit

**Reply by GiudGiud** [reply]
Ah thanks I did not know about these accommodations. Some modules are better than others wrt accepting other unit systems. Fluid properties doesn't do imperial units.

**Reply by rl3fz** [reply]
Very helpful, thank you both!
